aio/content/examples/dynamic-component-loader/src/app/ad-banner.component.ts

```diff
@@ -1,56 +1,45 @@
 // #docregion
-import { Component, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { Component, inject, OnDestroy, OnInit,  } from '@angular/core';
+import { CommonModule, NgComponentOutlet } from '@angular/common';
 
-import { AdDirective } from './ad.directive';
-import { AdItem } from './ad-item';
-import { AdComponent } from './ad.component';
+import { AdService } from './ad.service';
 
 @Component({
   selector: 'app-ad-banner',
-  // #docregion ad-host
+  standalone: true,
+  imports: [CommonModule, NgComponentOutlet],
+  // #docregion ng-container
   template: `
     <div class="ad-banner-example">
       <h3>Advertisements</h3>
-      <ng-template adHost></ng-template>
+      <ng-container *ngComponentOutlet="
+        currentAd.component;
```

Comment:
What do you think about using `async` pipe here and using dynamic imports in the service? :)
```suggestion
      <ng-container *ngComponentOutlet="
        currentAd.component | async;
```

Reply:
Hooray for lazy loading 👍